code/modules/religion/rites.dm

```diff
@@ -0,0 +1,223 @@
+/datum/religion_rites
+/// name of the religious rite
+	var/name = "religious rite"
+/// Description of the religious rite
+	var/desc = "immm gonna rooon"
+/// length it takes to complete the ritual
+	var/ritual_length = (10 SECONDS) //total length it'll take
+/// list of invocations said (strings) throughout the rite
+	var/list/ritual_invocations //strings that are by default said evenly throughout the rite
+/// message when you invoke
+	var/invoke_msg
+	var/favor_cost = 0
+
+/datum/religion_rites/New()
+	. = ..()
+	if(!GLOB?.religious_sect)
+		return
+	LAZYADD(GLOB.religious_sect.active_rites, src)
+
+/datum/religion_rites/Destroy()
+	if(!GLOB?.religious_sect)
+		return
+	LAZYREMOVE(GLOB.religious_sect.active_rites, src)
+	return ..()
+
+
+///Called to perform the invocation of the rite, with args being the performer and the altar where it's being performed. Maybe you want it to check for something else?
+/datum/religion_rites/proc/perform_rite(mob/living/user, atom/religious_tool)
+	if(GLOB.religious_sect?.favor < favor_cost)
+		to_chat(user, "<span class='warning'>This rite requires more favor!</span>")
+		return FALSE
+	to_chat(user, "<span class='notice'>You begin to perform the rite of [name]...</span>")
+	if(!ritual_invocations)
+		if(do_after(user, target = user, delay = ritual_length))
+			return TRUE
+		return FALSE
+	var/first_invoke = TRUE
+	for(var/i in ritual_invocations)
+		if(first_invoke) //instant invoke
+			user.say(i)
+			first_invoke = FALSE
+			continue
+		if(!ritual_invocations.len) //we divide so we gotta protect
+			return FALSE
```

Comment:
```suggestion
```

This check no longer is needed since length of `ritual_invocations` is already checked above.